extension/src/popup/ducks/settings.ts

```diff
@@ -335,6 +336,13 @@ export const settingsExperimentalModeSelector = createSelector(
   (settings) => settings.isExperimentalModeEnabled,
 );
 
+export const settingsSorobanSupportedSelector = createSelector(
+  settingsSelector,
+  (settings) =>
+    settings.isExperimentalModeEnabled ||
+    settings.networkDetails.network === NETWORKS.TESTNET,
```

Comment:
we should use this new selector (rather than isExperimentalModeEnabled) to determine if we should turn off/on Soroban features